### DISCUSSION TITLE
I need help with a problem I'm having developing an API for a chat page

### DISCUSSION THREAD

**Discussion by LucasNasc3000** [community/community#86751] [Programming Help] [answered]
### Body

I need the data to be returned in descending order, based on when it was created, but instead it is returned in alphabetical order according to the data in the "textcontent" column.
For some reason the order in which messages and other data related to them are displayed does not change even though I use "attributes" and "order" in the findAll() method
I'm using nodejs and the sequelize ORM.

Method to list all messages in API:

>  async index(req, res) {
    try {
      const message = await Text.findAll({
        attributes: ['id', 'msghour', 'created_at', 'useremail', 'textcontent'],
        order: [['textcontent', 'DESC'], ['useremail', 'DESC'],  ['created_at', 'DESC'],  ['id', 'DESC']],
      }); 
      return res.json(message);
    } catch (e) {
      return res.json(null);
    }
  }

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by DarrenChochian** [comment, marked as answer]
The order option is used to specify the sorting order for the retrieved records based on the specified columns.
try this:
async index(req, res) {
  try {
    const messages = await Text.findAll({
      attributes: ['id', 'msghour', 'created_at', 'useremail', 'textcontent'],
      order: [['created_at', 'DESC']],
    });
    return res.json(messages);
  } catch (e) {
    return res.json(null);
  }
}

**Reply by LucasNasc3000** [reply]
Thanks! Your solution worked. I also discovered that part of the problem is in the login middleware in the message routes file or in the git commands I use, because when I added all the files with the command "git add ." to send the changes to the server and deleted the middleware in the routes, the message display order was corrected.
